### PR TITLE
ui for creating persons/books and listing them

### PR DESCRIPTION
**Pull request by ishpaul777**
In this PR I have:

Created a console app that has methods that do the following:

- List all books.
- List all members.
- Create a person.
- Create a book.
- Create a rental.
- List all rentals for a given person's id.

run `ruby main. rb ` in the root of the project

### CODE REVIEW

File: classes/rental_records.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    person = members.select { |member| member.id == id }
+
+    if person.length.empty?
+      puts "OOPS Library# No me,ber found by Id -> #{id}"
+    else
+      puts 'OOPS Library# All Rental by the person-'
+      person[0].rentals.each do |r|
+        puts "Date: #{r.date}, Book -> #{r.book.title} by #{r.book.author}"
+      end
+    end
```
Comment (codepantha) [inline classes/rental_records.rb]: - [ ] There appears to be something wrong with this method which lists all rentals for a given person id. Please see error below and try to fix it 👇 
image

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: REQUIRES CHANGES ♻️ 

## Hi @ishpaul777 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights 🎯 
✔️ Descriptive PR
✔️ Good commit messages
✔️ Used correct GitFlow

## Required Changes ♻️
_Check the comments under the review._

- [ ] Please add the "Getting Started" section to your README in order to show users how to set up your project on their local machines. Kindly use this [template](https://github.com/microverseinc/readme-template) as a guide

- [ ] I noticed the link to the MIT license is broken. Kindly fix it as it is important for even open source software to be licensed 👍 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classes/rental_records.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    person = members.select { |member| member.id == id }
+
+    if person[0].length.empty?
+      puts "OOPS Library# No me,ber found by Id -> #{id}"
+    else
+      puts 'OOPS Library# All Rental by the person-'
+      person[0].rentals.each do |r|
+        puts "Date: #{r.date}, Book -> #{r.book.title} by #{r.book.author}"
+      end
```
Comment (RolandM99) [inline classes/rental_records.rb]: - [ ] Great job so far by trying to implement the changes requested by the previous reviewer. While implementing the changes, it seems that the errors change once we try to list all rentals.
image
There is an undefined method `length` which causes this error. Kindly, make sure that you fix this issue.

**Review by RolandM99 — CHANGES_REQUESTED** [review]
Hi @ishpaul777, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by danmainah — APPROVED** [review]
Hi @ishpaul777 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights

✔️ No linter errors
✔️ Github flow was correctly used
✔️  Good design

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**